### ISSUE TITLE
Work with the new `data-task` attribute

### ISSUE THREAD

**Issue by ryanjamurphy** [kepano/obsidian-minimal#8]
It'd be nice if Minimal had Minimal-aesthetic options for the new `data-task` attribute so that users could visually indicate e.g., dropped, deferred, or important tasks.

I haven't figured out the best way of editing the theme to achieve this, but I did find a way to make it work with @deathau's [checkbox styling](https://github.com/deathau/obsidian-snippets/blob/main/checkbox.css):

1. Remove the current checkbox checkmark, as it uses `background-image` which clashes with deathau's use of `-webkit-mask-image`. This means... delete [line 1936](https://github.com/kepano/obsidian-minimal/blob/159cbb8d3724a75166d582e8cb3d034d837107e3/obsidian.css#L1936) from Minimal's `obsidian.css`.
2. Remove `!important` from the `background-color` attribute of `.checklist-plugin-main .group .compact > .toggle .checked, .is-flashing input[type=checkbox]:checked, input[type=checkbox]:checked`—[line 1931](https://github.com/kepano/obsidian-minimal/blob/159cbb8d3724a75166d582e8cb3d034d837107e3/obsidian.css#L1931). This allows the different checkboxes to take on different colours.
3. Add the below CSS as a snippet and enable it. I removed the checkbox shaping in deathau's original CSS and added in a strikethrough effect on the text of dropped tasks.

```css
/* data-task attribute checkboxes from death_au, modified for minimal */
.markdown-preview-view .task-list-item-checkbox:checked {
  border: none;
  background-color: var(--interactive-accent);
}

.markdown-preview-view .task-list-item-checkbox:hover {
  background-color: var(--background-primary-alt);
}

.markdown-preview-view .task-list-item-checkbox:checked::before {
  position: absolute;
  color: white;
  text-align: center;
  font-weight: 900;
  line-height: 15px;
  width:12px;
  left:2px;
  right:2px;
}

.markdown-preview-view ul>li.task-list-item {
  font-weight: normal;
  color: var(--text-normal);
}

/* SVG check mark for done ('- [x]') */
.markdown-preview-view li[data-task="x"]>.task-list-item-checkbox:checked::before,
.markdown-preview-view li[data-task="X"]>.task-list-item-checkbox:checked::before {
  content: ' ';
  top: 2px;
  bottom: 2px;
  background-color: white;
  -webkit-mask-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' viewBox='0 0 12 12'%3E%3Cpolygon points='5.5 11.9993304 14 3.49933039 12.5 2 5.5 8.99933039 1.5 4.9968652 0 6.49933039'%3E%3C/polygon%3E%3C/svg%3E");
}

/* SVG chevron right for deferred/scheduled ('- [>]') */
.markdown-preview-view li[data-task=">"]>.task-list-item-checkbox:checked::before {
  content: '';
  top:2px;
  bottom:2px;
  background-color: white;
  -webkit-mask-image: url("data:image/svg+xml,%3Csvg viewBox='0 0 12 12' xmlns='http://www.w3.org/2000/svg'%3E%3Cpath d='M8.12 7L3.26667 11.76L4.57333 13.0667L10.7333 7L4.57333 0.933332L3.26667 2.24L8.12 7Z'/%3E%3C/svg%3E");
}

/* SVG line for cancelled/non-task ('- [-]') */
.markdown-preview-view li[data-task="-"]>.task-list-item-checkbox:checked::before {
  content: '';
  top:2px;
  bottom:2px;
  background-color: white;
  -webkit-mask-image: url("data:image/svg+xml,%3Csvg viewBox='0 0 12 12' xmlns='http://www.w3.org/2000/svg'%3E%3Crect x='0.5' y='6.5' width='13' height='1'/%3E%3C/svg%3E");
}
/* greyed-out background colour for cancelled/non-task ('- [-]') */
.markdown-preview-view li[data-task="-"]>.task-list-item-checkbox:checked {
  background-color: var(--text-faint);
}

/* strikethrough for dropped tasks */
[data-task="-"] {
    text-decoration: line-through !important;
}

/* '?' for question/more info task ('- [?]') */
.markdown-preview-view li[data-task="?"]>.task-list-item-checkbox:checked::before {
  content: '?';
  margin-left: .05em;
  margin-top: .05em;
}
/* text style for question/more info task ('- [?]') */
.markdown-preview-view ul>li[data-task="?"].task-list-item.is-checked {
  color: var(--text-normal);
  text-decoration: none;
}
/* yellow background colour for question/more info task ('- [?]') */
.markdown-preview-view li[data-task="?"]>.task-list-item-checkbox:checked{
  background-color: var(--background-modifier-accent);
}

/* '!' for important task ('- [!]') */
.markdown-preview-view li[data-task="!"]>.task-list-item-checkbox:checked::before {
  content: '!';
  margin-left: .05em;
  margin-top: .05em;
}
/* text style for important task ('- [!]') */
.markdown-preview-view ul>li[data-task="!"].task-list-item.is-checked {
  color: var(--text-normal);
  text-decoration: none;
  font-weight: 900;
}
/* red background colour for important task ('- [!]') */
.markdown-preview-view li[data-task="!"]>.task-list-item-checkbox:checked {
  background-color: var(--text-accent);
}

```

**Comment by dpmeltz** [comment]
Second - using SLRvb's checkboxes, that !important gets in the way. Looks like it's now on line 3360

**Comment by aventuraviper** [comment]
Is there a way to do this now? [Snippetor](https://github.com/ebullient/obsidian-snippetor) seems to be able to write CSS snippets that override theme checkboxes gracefulyl without editing theme files.

**Comment by eleanorkonik** [comment]
Adding my support for this, snippetor works but I'd love to see a version that feels "native" to Minimal.

**Comment by kepano** [comment]
This is probably a dumb question, but do you all use Source view to manually edit your alternate checkbox styles or is there a plugin/hotkeys you use to the various checkbox types?

**Comment by eleanorkonik** [comment]
I just type it in. In Live Preview it works like any other rendered thing and goes back to "source" when your cursor is in that section. 

I think somebody built a plugin to give hotkeys for it but I don't bother and if it does exist I forget what it's called.

**Comment by aventuraviper** [comment]
> I just type it in. In Live Preview it works like any other rendered thing and goes back to "source" when your cursor is in that section.
> 
> I think somebody built a plugin to give hotkeys for it but I don't bother and if it does exist I forget what it's called.

Are you thinking of [Task Collector](https://github.com/ebullient/obsidian-task-collector) (by [Snippetor creator](https://github.com/ebullient/obsidian-snippetor))? Not sure it has hotkeys but it has commands for marking tasks and can be setup with various different types (e.g. all the ones now in [Primary](https://user-images.githubusercontent.com/90643719/156046290-afdc074c-ea07-4f74-891f-112652ba0af9.png)).

**Comment by kepano** [comment]
This is now available as of 5.1.8
